Scripts/Python/xRegisterAge.py

```diff
@@ -0,0 +1,120 @@
+(or a modified version of those libraries),
+containing parts covered by the terms of the Bink SDK EULA, 3ds Max EULA,
+PhysX SDK EULA, DirectX SDK EULA, OpenSSL and SSLeay licenses, IJG
+JPEG Library README, Windows Media SDK EULA, or QuickTime SDK EULA, the
+licensors of this Program grant you additional
+permission to convey the resulting work. Corresponding Source for a
+non-source form of such a combination shall include the source code for
+the parts of OpenSSL and IJG JPEG Library used as well as that of the covered
+work.
+
+You can contact Cyan Worlds, Inc. by email legal@cyan.com
+ or by snail mail at:
+      Cyan Worlds, Inc.
+      14617 N Newport Hwy
+      Mead, WA   99021
+
+ *==LICENSE==* """
+
+from Plasma import *
+from PlasmaTypes import *
+from PlasmaKITypes import *
+
+act = ptAttribActivator(1, "Act: Buttons")
+resp = ptAttribResponder(2, "Resp: Button Push")
+ageFileName = ptAttribString(3, "age filename")
+ageInstanceName = ptAttribString(4, "age instance name")
+
+class xRegisterAge(ptResponder):
+
+    def __init__(self):
+        # run parent class init
+        ptResponder.__init__(self)
+        self.id = 5008
+        version = 1
+        self.version = version
+        PtDebugPrint("__init__xRegisterAge v.", version)
+        
+    def OnNotify(self,state,id,events):
+        if state and PtFindAvatar(events) == PtGetLocalAvatar():
+            if id==act.id:
+                if not self.IFindAgeLinkInFolder(ageFileName.value):
+                    if resp.value:
+                        resp.run(self.key,events=events)
+                    PtSendKIMessage(kKILocalChatStatusMsg, PtGetLocalizedString("Global.Messages.AgeBookGet", [ageInstanceName.value]))
+                    self.CheckAndRegisterAge(ageFileName.value, ageInstanceName.value)
+
+    def IFindAgeLinkInFolder(self, ageName):
+        vault = ptVault()
+        folder = vault.getAgesIOwnFolder()
+        ageName = ageName.lower()
+        contents = folder.getChildNodeRefList()
+        for content in contents:
+            link = content.getChild().upcastToAgeLinkNode()
+            if link is not None:
+                info = link.getAgeInfo()
+            else:
+                link = content.getChild()
+                info = link.upcastToAgeInfoNode()
+
+            if info.getAgeFilename().lower() == ageName:
+                return link
+        return None
+
+    def CheckAndRegisterAge(self, ageFileName, ageInstanceName):
+        PtDebugPrint("xRegisterAge.CheckAndRegisterAge(): ",ageFileName,"; ",ageInstanceName)
+        vault = ptVault()
+        ageStruct = ptAgeInfoStruct()
+        ageStruct.setAgeFilename(ageFileName)
+        ageLinkNode = vault.getOwnedAgeLink(ageStruct)
+        if not ageLinkNode:
+            info = ptAgeInfoStruct()
+            info.setAgeFilename(ageFileName)
+            info.setAgeInstanceName(ageInstanceName)
+
+            playerName = PtGetClientName()
+            ageGuid = PtGuidGenerate()
```

Comment:
I'm pretty sure that if you don't set the guid, then the server will generate one for you.

Reply:
I took this function from https://github.com/H-uru/Plasma/blob/af1395d8b6da492817b78224948966238db0c2c3/Scripts/Python/xLinkingBookGUIPopup.py#L1045